New package: 2D Alpha Wrapping

MaelRL opened the pull request:
## Summary of Changes

This component takes a 2D triangle soup, edge soup, or point set as input, and generates a simple polygon soup that contains the input (watertight, intersection-free, and combinatorially 1-manifold). The algorithm proceeds by shrink-wrapping and refining a 2D Delaunay triangulation loosely bounding the input. Two user-defined parameters, alpha and offset, offer control over the maximum size of cavities where the shrink-wrapping process can enter, and the tightness of the final surface mesh to the input, respectively. Once combined, these parameters provide a means to trade fidelity to the input for complexity of the output.

See Alpha_wrap_2.h for TODOS.

## Release Management

Depends on https://github.com/CGAL/cgal/pull/9096

* Affected package(s): `Alpha_wrapping_2`, `Alpha_wrapping_3`
* Issue(s) solved (if any): -
* Feature/Small Feature (if any): [Alpha_wrap_2](https://cgalwiki.geometryfactory.com/CGAL/Members/wiki/Features/Alpha_wrap_2)
* License and copyright ownership:

MaelRL committed:
Make AABB_traits typedef public

It's public in _3 and it is needed for Alpha_wrap_2

MaelRL committed:
Add Kernel_23::Compare_squared_radius_2

MaelRL committed:
Add 2D Robust_weighted_circumcenter filtered traits

MaelRL committed:
Add Delaunay and Delaunay_with_circumcenter_2 face bases

MaelRL committed:
Fix compilation behind expensive assertion

MaelRL committed:
Fix typo

MaelRL committed:
Add trivial Point_2 / Point_2 projection overload

MaelRL committed:
Add alpha_wrap_2 (wip)

afabri committed:
Add license file

MaelRL committed:
3D > 2D

MaelRL committed:
Merge remote-tracking branch 'cgal/main' into AW2-GF

MaelRL committed:
Merge remote-tracking branch 'soesau/Intersection-Circle_Sphere_consistency-GF' into AW2-GF

MaelRL committed:
Clean indentation

MaelRL committed:
Add a label->string helper

MaelRL committed:
Fix failing to detect that some gates are zombies

See comment in the code.

MaelRL committed:
Clean debug code

MaelRL committed:
AW2: Use WKT everywhere

MaelRL committed:
Add a clear() function

MaelRL committed:
Add more accessors from the wrapper

MaelRL committed:
Rewrite the boundary extraction for AW2

MaelRL committed:
Return the refinement rule in Steiner computation + convenience overload

MaelRL committed:
Remove irrelevant oracle files

There is no point having triangles in 2D: we are only interested
in their borders, for which the segment oracle is sufficient

MaelRL committed:
Fix linking error

MaelRL committed:
Add missing region label for boundary extraction

MaelRL committed:
Fix broken clear function: can't && voids

MaelRL committed:
Rewrite AW2 API

+ fix storage oracle bug

MaelRL committed:
Add code to dump labels in debug code

MaelRL committed:
Add clear() functions for AW2 oracles

MaelRL committed:
Add 2 new examples + update old ones

MaelRL committed:
Move AW2 package on package list

MaelRL committed:
Update comment: AW2 also needs the precondition-less version

MaelRL committed:
Minor update to a comment: those NPs are used by AW3 *and* AW2

MaelRL committed:
Add validation functions for AW2

MaelRL committed:
Add test for AW2 on point sets

MaelRL committed:
Add basic doc

MaelRL committed:
Misc cleaning

MaelRL committed:
Add code to compute gate properties at create time + cache them (for demo)

MaelRL committed:
Mirror AW2 improvements into AW3

MaelRL committed:
Add demo for AW2

MaelRL committed:
Fix test compilation

MaelRL committed:
The doc of AW3 depends on AW2

MaelRL committed:
Add a missing helper function

MaelRL committed:
Fix bad copy paste

MaelRL committed:
Add .obj polyline reading to CGALlab

MaelRL committed:
Add missing namespaces

MaelRL committed:
Work on user manual

MaelRL committed:
Work on user manual

MaelRL committed:
Minor ref manual improvements

MaelRL committed:
Add a test for polyline inputs

MaelRL committed:
Misc minor changes

MaelRL committed:
Update todos

MaelRL committed:
Fix fig labels in AW23

MaelRL committed:
Fix paranthesis placements for doc

MaelRL committed:
Add dependencies

MaelRL committed:
Fix case

MaelRL committed:
Scale up figure a tiny bit

MaelRL committed:
Remove needless includes

MaelRL committed:
Add some concepts

Kinda pointless, because these concepts are almost empty shells

MaelRL committed:
Merge remote-tracking branch 'soesau/Intersection-Circle_Sphere_consistency-GF' into AW2-GF

MaelRL committed:
Remove obsolete todo task

MaelRL committed:
Fix circle tracing + true R1 implementation

MaelRL committed:
Add a basic sphere tracing function for AW3's AABB oracles

But (not sure why atm) this + R1 is significantly slower for 3D
so leaving it unused for now.

MaelRL committed:
Fix debug code compilation

MaelRL committed:
Add Polygon_2 / GeneralPolygonWithHoles overloads + API tests

MaelRL committed:
Document new Kernel concept requirements

MaelRL committed:
Misc T3 doc fixes

MaelRL committed:
Document Robust_circumcenter_filtered_traits_3<K>

MaelRL committed:
Document Delaunay_face_base_[with_circumcenter_]2

MaelRL committed:
Document Robust_[weighted]_circumcenter_filtered_traits_2

MaelRL committed:
Document DelaunayTriangulationFaceBase_2

MaelRL committed:
Update todos

MaelRL committed:
Add a word

MaelRL committed:
Misc doc improvements across T2

MaelRL committed:
Instantiate by > with across CGAL

MaelRL committed:
Doc fixes for CompareSquaredRadius_23

MaelRL committed:
Document default named parameters

MaelRL committed:
Add polygon validity remarks

MaelRL committed:
Do not offer GeneralPolygonWithHoles, but only CGAL::Polygon_with_holes_2

No support for curved inputs

MaelRL committed:
AW2 user manual fixes

MaelRL committed:
Add an internal but unused polygon chain overload to the segment oracle

MaelRL committed:
Fix warning

MaelRL committed:
Misc minor ref manual fixes

MaelRL committed:
Add undocumented np for polygonal chains (AW2)

MaelRL committed:
Handle null length voronoi dual edges

MaelRL committed:
clean test

MaelRL committed:
Fix warnings

MaelRL committed:
Eight is four in 2D

MaelRL committed:
Actually avoid the precondition of orientation()

MaelRL committed:
Use .bbox() to get the bounding box of a primitive if no pmap is provided

The doc says that .bbox() must exist and Compute_bbox_2 is not part
of the AABBGeomTraits_2 concept.

MaelRL committed:
Introduce AlphaWrapTraits_2

MaelRL committed:
Add missing operators

MaelRL committed:
Fix macro name

MaelRL committed:
Merge remote-tracking branch 'mine/Polygon_repair-Fix_is_valid-GF' into AW2-GF

MaelRL committed:
Remove zealous precondition

MaelRL committed:
Make it possible to use Cartesian_converter with partial kernels...

... + use functors to extract and build kernel objects.

MaelRL committed:
Simplify the code that constructs the initial loose bbox

MaelRL committed:
Replace functor to minimize traits requirements

MaelRL committed:
Simplify orientation of circumcenter predicates

In 2D, we have an equivalence between a side_of(circumcenter, opposite
edge) and the opposite angle.

MaelRL committed:
Update todo list

MaelRL committed:
Add an AW2 Traits API test

MaelRL committed:
Update AWTraits2

MaelRL committed:
Merge branch 'main' into AW2-GF

MaelRL committed:
Do not ignore seeds outside of the bbox (wip)

Cavity initilization is currently still broken, so the proper fix
is not only to ignore them, but to also mark properly initialize
when a seed is outside the bbox.

MaelRL committed:
Review fixes

MaelRL committed:
Aff_transformation_23 do not have a kernel functor

MaelRL committed:
Do not copy the oracle

MaelRL committed:
Fix oracle stacks with empty tops

How did that even work for AW3?...........

MaelRL committed:
Fix AW2 demo on hand drawn only inputs

MaelRL committed:
Remove debug macros

MaelRL committed:
User manual improvements after Sven's review

MaelRL committed:
Fix concept in AW2 user manual

MaelRL committed:
Remove obsolete figure

MaelRL committed:
Fix functor name in robust_weighted_CC_traits_2

MaelRL committed:
Add missing reserve() to AW2/AW3 oracles

MaelRL committed:
Fix inverted true/false in validation function

MaelRL committed:
Fixed not propagating named parameters for some AW2 overloads

MaelRL committed:
Factorize + enhance some test functions

MaelRL committed:
Add a test for AW2 manifoldness

MaelRL committed:
Actually return false when the validation fails

MaelRL committed:
Fix bad copy paste

MaelRL committed:
Make it possible to test 3D STL data in AW2 tests

MaelRL committed:
Add an example of interior wrapping

MaelRL committed:
Update todos

MaelRL committed:
Misc cleaning

MaelRL committed:
Add data set

chris-gagneraud-ctct commented on Stream_support/include/CGAL/IO/OBJ.h:
```diff
@@ -149,13 +151,70 @@ bool read_OBJ(std::istream& is,
         return false;
       }
     }
+    else if(s == "l")
```
Hi, I actually asked recently about polyline support in OBJ files. 
See https://github.com/CGAL/cgal/issues/9311
Any chance you could add polyline support to `write_OBJ` too?

MaelRL replied:
I have added an internal version with polyline support in [df67690](https://github.com/CGAL/cgal/pull/8856/commits/df6769092e06be3a31632dc957b2e1af19f86984).

MaelRL committed:
Add an undocumented OBJ polyline writer

MaelRL committed:
Fix compilation issues

MaelRL committed:
Fix warning

sloriot committed:
fix functor names

sloriot committed:
Merge remote-tracking branch 'cgal/main' into 'MaelRL/AW2-GF'

sloriot committed:
heart needs a larger ratio

sloriot committed:
fix order of parameters

afabri committed:
remove unused using statements

afabri committed:
Polygon and Polyline without _2 cause trouble with <windows.h> included

afabri committed:
Another _2

janetournois committed:
fix compilation error on msvc

msvc has an issue with constexpr lambdas with templates
so we replace it by a constexpr function

janetournois committed:
complete previous commit

MSVC does not handle properly constexpr lambdas with templates

sloriot committed:
fix path

afabri committed:
Fix std::min VC problem

afabri committed:
Link against Opengl32

afabri committed:
Link with OpenGL::GL

sloriot committed:
workaround warnings for macOs

sloriot committed:
fix conversion warning detected with MSVC

sloriot committed:
do not test data set removed

sloriot committed:
fix import
